src/irr/asset/CSmoothNormalGenerator.cpp

```diff
@@ -76,22 +84,127 @@ void CSmoothNormalGenerator::processConnectedVertices(asset::ICPUMeshBuffer* buf
 
 		for (size_t j = i+1; j < vertices.size(); j++)
 		{
-			if (buffer->getPosition(processedVertex.indexOffset).x == buffer->getPosition(vertices[j].indexOffset).x &&
-				buffer->getPosition(processedVertex.indexOffset).y == buffer->getPosition(vertices[j].indexOffset).y &&
-				buffer->getPosition(processedVertex.indexOffset).z == buffer->getPosition(vertices[j].indexOffset).z)
+			if (compareVertexPosition(processedVertex.position, vertices[j].position, epsilon))
 			if (processedVertex.parentTriangleFaceNormal.dotProductAsFloat(vertices[j].parentTriangleFaceNormal) > cosOfCreaseAngle)
 			{
 				processedVertex.normal += vertices[j].parentTriangleFaceNormal * vertices[j].wage;
 				vertices[j].normal += processedVertex.parentTriangleFaceNormal * processedVertex.wage;
 			}
+			
 		}
 		
 		processedVertex.normal = core::normalize(processedVertex.normal);
-		if (!buffer->setAttribute(processedVertex.normal, asset::E_VERTEX_ATTRIBUTE_ID::EVAI_ATTR3, processedVertex.indexOffset))
-			_IRR_DEBUG_BREAK_IF(true);
+		buffer->setAttribute(processedVertex.normal, asset::E_VERTEX_ATTRIBUTE_ID::EVAI_ATTR3, processedVertex.indexOffset);
 	}
+
+}
+
+#pragma endregion
+
+
+#pragma region hashing
+
+CSmoothNormalGenerator::VertexHashMap::VertexHashMap(size_t _hashTableSize, float _cellSize)
+	:hashTableSize(_hashTableSize), cellSize(_cellSize)
+{
+	hashTable.reserve(hashTableSize);
+
+	for (int i = 0; i < hashTableSize; i++)
+		hashTable.emplace_back();
```

Comment:
you know you could just use a std::vertex::resize instead of a reserve+emplace loop